src/Dictionaries/getDictionaryConfigurationFromAST.cpp

```diff
@@ -614,6 +615,15 @@ getDictionaryConfigurationFromAST(const ASTCreateQuery & query, ContextPtr conte
 
     checkPrimaryKey(all_attr_names_and_types, pk_attrs);
 
+    /// If the pk size is 1 and pk's DataType is not native uint(UInt8~UInt64), we should convert to complex,
+    /// because the data type of Numeric key(simple layout) is UInt64.
+    if ((pk_attrs.size() > 1 || (pk_attrs.size() == 1 && !WhichDataType(DataTypeFactory::instance().get(all_attr_names_and_types.find(pk_attrs[0])->second.type)).isNativeUInt()))
```

Comment:
I guess the old code uses "is convertible to UInt64", so I think the same should be used here, @kitaisreal what do you think ?
And converted later to something else separately (if at all)